Tasks/JenkinsQueueJob/jenkinsqueuejobtask.ts

```diff
@@ -9,196 +9,87 @@ import tl = require('vsts-task-lib/task');
-var jenkinsExecutableUrl;
-
-function trackJobQueued(queueUri: string) {
-    tl.debug('Tracking progress of job queue: ' + queueUri);
-    request.get({ url: queueUri }, function callBack(err, httpResponse, body) {
-        if (err) {
-            tl.setResult(tl.TaskResult.Failed, err);
-        } else if (httpResponse.statusCode != 200) {
-            failReturnCode(httpResponse, 'Job progress tracking failed to read job queue');
-        } else {
-            var parsedBody = JSON.parse(body);
-            // canceled is spelled wrong in the body with 2 Ls (checking correct spelling also in case they fix it)
-            if (parsedBody.cancelled || parsedBody.canceled) {
-                tl.setResult(tl.TaskResult.Failed, 'Jenkins job canceled.');
-            }
-            var executable = parsedBody.executable;
-            if (!executable) {
-                // job has not actually been queued yet, keep checking
-                setTimeout(function () {
-                    trackJobQueued(queueUri);
-                }, captureConsolePollInterval);
-            } else {
-                jenkinsTaskName = parsedBody.task.name;
-                jenkinsExecutableNumber = parsedBody.executable.number;
-                jenkinsExecutableUrl = parsedBody.executable.url;
-                console.log('Jenkins job started: ' + jenkinsExecutableUrl);
-
-                if (captureConsole) {
-                    // start capturing console
-                    captureJenkinsConsole(0);
-                } else {
-                    // no console option, just create link and finish
-                    createLinkAndFinish(tl.TaskResult.Succeeded, 'Queued', 'Jenkins job successfully queued: ' + jenkinsExecutableUrl);
-                }
-            }
-        }
-    });
-}
+import job = require('./job');
+import Job = job.Job;
+import jobqueue = require('./jobqueue');
+import JobQueue = jobqueue.JobQueue;
+import util = require('./util');
 
-function createLinkAndFinish(result, jobStatus: string, resultMessage: string) {
-    var tempDir = shell.tempdir();
-    var linkMarkdownFile = path.join(tempDir, 'JenkinsJob_' + jenkinsTaskName + '_' + jenkinsExecutableNumber + '.md');
-    tl.debug('jenkinsLink: ' + linkMarkdownFile);
-    var summaryTitle = 'Jenkins ' + jenkinsTaskName + ' - ' + jenkinsExecutableNumber + ' - ' + jobStatus;
-    tl.debug('summaryTitle: ' + summaryTitle);
-    var markdownContents = '[' + jenkinsExecutableUrl + '](' + jenkinsExecutableUrl + ')';
-    fs.writeFile(linkMarkdownFile, markdownContents, function callBack(err) {
-        if (err) {
-            //don't fail the build -- there just won't be a link
-            console.log('Error creating link to Jenkins job: ' + err);
-        } else {
-            console.log('##vso[task.addattachment type=Distributedtask.Core.Summary;name=' + summaryTitle + ';]' + linkMarkdownFile);
-        }
-        tl.setResult(result, resultMessage);
-    });
-}
+export class TaskOptions {
```

Comment:
Can additional type information be applied to the members of this class where missing (such as `serverEndpointUrl`)?